src/GraphQL/Utilities/SchemaPrinter.cs

```diff
@@ -338,21 +347,25 @@ private string FormatInputObjectValue(object value, IInputObjectGraphType input)
             var sb = new StringBuilder();
             sb.Append("{ ");
 
-            foreach (var field in input.Fields)
+            foreach (var field in input.Fields.OrderBy(Options.Comparer?.FieldComparer(input)))
             {
                 string propertyName = field.GetMetadata<string>(ComplexGraphType<object>.ORIGINAL_EXPRESSION_PROPERTY_NAME) ?? field.Name;
-                PropertyInfo propertyInfo;
 
-                try
-                {
-                    propertyInfo = value.GetType().GetProperty(propertyName, BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.Instance);
-                }
-                catch (AmbiguousMatchException)
+                if (!(value is IDictionary<string, object?> dic && dic.TryGetValue(propertyName, out var propertyValue)))
```

Comment:
This is a bug fix: SchemaPrinter would not print a schema with a default input object value that was parsed by Schema.For.

Reply:
Could you explain in more details?

Reply:
Given this schema:

```gql
input SomeInput1 {
  field1: SomeInput2 = { age: 22 }
}
input SomeInput2 {
  age: Int
}
```

When creating the schema in code-first, the default value would likely be `new SomeInput2 { age = 22 }`.  However, when parsing the above text via `Schema.For`, the default value is `new Dictionary<string, object?> { { "age", 22 } }`.  The default value is valid and `GetArgument<SomeInput2>` will coerce it to the proper type.  However, the schema printer currently expects a code-first object and does not expect a dictionary.  This code change fixes that problem so it can be printed.

Reply:
So the if branch is for code-first "object" value to take some property from it. And then we have either that value or just "plain" value from dictionary. Right? Worth to add comments.

Reply:
Comments added; please review.  Had to update logic because found two more bugs.